### PR TITLE
feat: support customize route prefix

### LINKED ISSUE #1713: set a custom url prefix for the dashboard
# Feature request

## Please describe your feature

When Using apisix-dashboard behind a proxy (like nginx), which serves other services。Set a base url for apisix-dashboard will allow nginx routes to the correct place

### PR DESCRIPTION
**Pull request by jwrookie**
Please answer these questions before submitting a pull request, **or your PR will get closed**.

**Why submit this pull request?**

- [ ] Bugfix
- [x] New feature provided
- [ ] Improve performance
- [ ] Backport patches

**What changes will this PR take into?**
Support customize route prefix

**Related issues**

fix/resolve #1713

**Checklist:**

- [x] Did you explain what problem does this PR solve? Or what new features have been added?
- [x] Have you added corresponding test cases?
- [ ] Have you modified the corresponding document?
- [x] Is this PR backward compatible? If it is not backward compatible, please discuss on the mailing list first

### CODE REVIEW

File: api/test/shell/cli_test.sh [hunk 1]
```diff
@@ -483,6 +483,39 @@ stop_dashboard() {
...
+
+  # check api
+  run curl http://127.0.0.1:9000/test/apisix/admin/user/login -H "Content-Type: application/json" -d '{"username":"admin", "password": "admin"}'
+  respCode=$(echo "$output" | sed 's/{/\n/g'| sed 's/,/\n/g' | grep "code" | sed 's/:/\n/g' | sed '1d')
+  echo $respCode
+  [ "$respCode" = "0" ]
+
+  # 404
+  result=$(curl "http://127.0.0.1:9000/apisix/admin/user/login")
+  [ "$result" = "hi~" ]
```
Comment (starsz) [inline api/test/shell/cli_test.sh:510]: Can this pass?

Comment (starsz) [reply api/test/shell/cli_test.sh:510]: Missing check of 404.

Comment (jwrookie) [reply api/test/shell/cli_test.sh:510]: 404 was handled

https://github.com/apache/apisix-dashboard/blob/a2fe71d0644927eb71ef3dd06b902ea9f4ce81f6/api/internal/route.go#L52-L65

**Review by starsz — APPROVED** [review]

**Review by starsz — APPROVED** [review]

**Comment by jwrookie** [comment]
@bzp2010 @nic-chen  have a check , thanks!

**Comment by bzp2010** [comment]
Hi, @jwrookie.

I noticed that this PR does not include front-end related changes, so I don't think configuring it will work. 

The current front-end code cannot dynamically adapt to changes in the root directory (baseURL), it can only be changed at the compiler, so I don't think it will work properly just by changing it.

**Comment by jwrookie** [comment]
> Hi, @jwrookie.
> 
> I noticed that this PR does not include front-end related changes, so I don't think configuring it will work.
> 
> The current front-end code cannot dynamically adapt to changes in the root directory (baseURL), it can only be changed at the compiler, so I don't think it will work properly just by changing it.

ok, let me check

**Review by nic-chen — APPROVED** [review]

**Review by SkyeYoung — APPROVED** [review]

**Comment by macdoor** [comment]
This feature is very useful. Please keep moving forward

**Pull request closed without merging**